Use new submit for review API

dalemyers opened the pull request:
The old version of this API has been deprecated and will no longer work from June 9th: https://developer.apple.com/documentation/appstoreconnectapi/post-v1-appstoreversionsubmissions 

This moves to the new API. 

Note: I haven't been able to test this yet. @JasperGuo / @Cokile if you have a test app you can test this with, it would be appreciated!

Cokile commented:
FYI: we tried to migrate to use the new API before, and fixed some bugs: #46, #47, but the API seems not work as expected, so we reverted it back: #50

dalemyers commented on asconnect/version_client.py:
```diff
@@ -454,14 +456,13 @@ def submit_for_review(
-                            "appStoreVersion": {
-                                "data": {"type": "appStoreVersions", "id": version_id}
-                            }
+                            "app": {"data": {"type": "appStoreVersions", "id": version_id}}
```
Great catch!

dalemyers replied:
Looks like I've also got version id instead of app id.

dalemyers replied:
Ok, I've fixed both of these.

dalemyers commented:
> FYI: we tried to migrate to use the new API before, and fixed some bugs: #46, #47, but the API seems not work as expected, so we reverted it back: #50

That's somewhat annoying. It's been a year though, so maybe things have improved? The API we were using is going away, so I don't think we have much choice but to go with this approach and try and deal with issues if/when they come up?
